src/main/java/rx/internal/operators/OperatorZip.java

```diff
@@ -181,9 +180,7 @@ public void request(long n) {
         private final CompositeSubscription childSubscription = new CompositeSubscription();
 
         @SuppressWarnings("unused")
-        volatile long counter;
-        @SuppressWarnings("rawtypes")
-        static final AtomicLongFieldUpdater<Zip> COUNTER_UPDATER = AtomicLongFieldUpdater.newUpdater(Zip.class, "counter");
+        AtomicLong counter = new AtomicLong();
```

Comment:
Alternatively, `Zip` could extend `AtomicLong` (or `AtomicInteger`) and we save in memory and updater overhead as well.